Functions/QualityControl/xASL_qc_CollectParameters.m

```diff
@@ -113,7 +113,26 @@
 %% -----------------------------------------------------------------------------------------------
 %% Save QC output
 QC_Path = fullfile(x.D.ROOT, x.SUBJECTS{iSubject}, ['QC_collection_' x.SUBJECTS{iSubject} '.json']);
-xASL_delete(QC_Path);
+
+% Load QC.json
+if xASL_exist(QC_Path, 'file')
```

Comment:
Somehow this part is not collaborating with the part above, where QC_path is already loaded etc.

Reply:
Actually. Above (especially lines 57-67), we create the new QC parameters. 
And here, we load the old QC parameters from file so that we can merge old and new. Because previously, we were deleting old (all previous sessions) and saving new (the current session), so this had to be adapted to be able to keep more than the current ASL session.

Is that fine, shall I add explanatory comments in the code?

Reply:
Are you sure? In line 57-67 -> xASL_qc_CollectQC_ASL we load the new fields x.Output.ASL.(SessionID), but we also throw away the old field x.Output.ASL content (you wrote a warning there).

This assumes that we already loaded the QC_Path previously, which is true, we load the QC json at the beginning of the ASL module, then populate it within each wrapper/submodule, and at the end we write it again. This QC collection part here is a temporary solution, in fact all QC parameters should be written to memory x.Output.ASL.(SessionID) in the respective wrappers/submodules that create them. (just explaining why we should load QC json at the beginning of the ASL module and save at the end of the ASL module)

So by definition, we have already loaded the QC_Path, and here we only save it, for which we ensure to first delete it. So you have to revert this edit.
Makes sense?

Reply:
Not really making sense. The problem is, if we keep everything in the memory, we will loose this memory after each session. And if we have processed sessions 1 and 2 previously. And now added session 3, we will need to load this from the path again.

Or your comment makes sense, but you don't realize that with each module run for a session, we clean x-struct, so that's why reloading is needed.

Or do I really get that wrong? I have been testing it and it seemed to be necessary to have this loading here - by a practical test.

Perhaps you can try to test it yourself and have some pre-processed and some new ASL sessions and see if all sessions would have been present. It seems to me they wouldn't be if I revert this edit...

Reply:
OK let me test this then. Removing old sessions, cleaning up, etc, should be done at the beginning of xASL_module_ASL, then it gets populated throughout all submodules, and at the end we save it.

But you're right that if ExploreASL crashes in the meantime, it should have everything saved to the disk. Not sure how we deal with this now; at the start all QC was indeed loaded and saved only in the last QC submodule. But we now do in fact output e.g. some Tanimoto parameters in registration, not sure how/if these get saved on disk properly.

Reply:
I understand. Currently, we NEED to load and save QC from other session in each xASL_module_ASL. But you are right that we might not need to do that at the end, but that it could be done at the start of the module.

If the real issue is about loading at start and saving at the end. Instead of loading in the middle and saving at the end, then we can most likely move it without any consequences.

If you confirm that this is the problem, then I can investigate and change the code. Most likely, this is going to work, so no reason not to do that :)

Reply:
OK. All clear. I will check and implement it that way.

Reply:
What to do

- [ ] xASL_adm_CleanUpBeforeRerun - only clean QC of the current session.
- [ ] Output_im should also have subsessions for ASL
- [ ] xASL_module_ASL part C - only remove the current session.
- [ ] xASL_module_ASL before part C - add a function to clean old QC - move the code from the other function.
- [ ] xASL_module_ASL after part C - properly read the QC from old modules and add to x-struct.

Reply:
All done. We just need to decide if we move Output_im for ASL to subsessions. We probably should.

Reply:
yes, and the current new "old cleanup" function should be incorporated in xASL_adm_CleanUpBeforeRerun section 9

Reply:
Output_im fixed in 0ff0f041
About integrating "old cleanup", see my comment below.